vispy/visuals/volume.py

```diff
@@ -687,6 +687,11 @@ def cmap(self):
     def cmap(self, cmap):
         self._cmap = get_colormap(cmap)
         self.shared_program.frag['cmap'] = Function(self._cmap.glsl_map)
+        self.shared_program['texture2D_LUT'] = (
+            self.cmap.texture_lut()
+            if (hasattr(self.cmap, 'texture_lut'))
+            else None
+        )
```

Comment:
That `hasattr` isn't very pretty :)  I see that it's a pattern used in several places in Vispy. It looks like `Colormap` always has `texture_lut()` (and it may return None), but `BaseColormap` does not. @djhoese what about we add a stub `texture_lut()` method to `BaseColormap` that simply returns None? Under the assumptions that all user-given colormaps inherit from `BaseColormap`, we can get rid of these `hasatrr`...

Reply:
I'm ok with that...in another PR